### PR TITLE
feat(auth): auth refresh tokens

### LINKED ISSUE #4427: Add authentication refresh mechanism into Relay network

### PR DESCRIPTION
**Pull request by mikeldking**
resolves #4427

### CODE REVIEW

File: app/src/authFetch.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+      // Retry the original request
+      return fetch(input, init);
+    }
+  }
+  throw new Error("An unexpected error occurred while fetching data");
+}
+
+let refreshPromise: Promise<Response> | null = null;
+
+async function refreshTokens(url: string = REFRESH_URL): Promise<Response> {
```
Comment (Parker-Stafford) [inline app/src/authFetch.ts:39]: why is this not nullable yet has a fallback? also it's not passed in above can we just use the constant here everytime instead?

File: app/src/authFetch.ts [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  if (refreshPromise) {
+    // There is already a refresh request in progress, so we should wait for it
+    return refreshPromise;
+  }
+  // This function should make a request to the server to refresh the access token
+  // eslint-disable-next-line no-console
+  console.log("Refreshing tokens");
```
Comment (Parker-Stafford) [inline app/src/authFetch.ts:46]: do we need this log?

File: app/src/authFetch.ts [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    if (!response.ok) {
+      // for now force redirect to login page. This could re-throw with a custom error
+      // But for now, we'll just redirect
+      window.location.href = "/login";
+    }
+    // Clear the refreshPromise so that future requests will trigger a new refresh
+    refreshPromise = null;
+    return response;
+  });
+  return refreshPromise;
```
Comment (Parker-Stafford) [inline app/src/authFetch.ts:59]: does this need to return? doesn't seem to be used up above

File: app/src/authFetch.ts (hunk 2 above)
Comment (mikeldking) [reply app/src/authFetch.ts:46]: Nope

File: app/src/authFetch.ts (hunk 3 above)
Comment (mikeldking) [reply app/src/authFetch.ts:59]: you can't await it otherwise. Can you clarify your question a bit?

File: app/src/authFetch.ts (hunk 1 above)
Comment (mikeldking) [reply app/src/authFetch.ts:39]: Yeah we can - I was just trying to not hard code things but I guess it's kinda stupid

File: src/phoenix/auth.py [hunk 4]
```diff
@@ -192,11 +192,11 @@ def validate(
 """The algorithm to use for the JSON Web Token."""
 PHOENIX_ACCESS_TOKEN_COOKIE_NAME = "phoenix-access-token"
 """The name of the cookie that stores the Phoenix access token."""
-PHOENIX_ACCESS_TOKEN_MAX_AGE = timedelta(minutes=10)
+PHOENIX_ACCESS_TOKEN_MAX_AGE = timedelta(seconds=10)
```
Comment (Parker-Stafford) [inline src/phoenix/auth.py:195]: sould both of these be seconds?

**Review by Parker-Stafford — APPROVED** [review]

File: src/phoenix/auth.py (hunk 4 above)
Comment (mikeldking) [reply src/phoenix/auth.py:195]: Oh I shouldn't have checked this in

File: app/src/authFetch.ts (hunk 3 above)
Comment (Parker-Stafford) [reply app/src/authFetch.ts:59]: ahh right, sorry was thinking it was returning the response of the promise not the promise

**Pull request merged**